src/field_type.cpp

```diff
@@ -176,6 +176,9 @@ void field_types::finalize_all()
 {
     set_field_type_ids();
     all_field_types.finalize();
+    for( const field_type &fd : all_field_types.get_all() ) {
```

Comment:
The line above does finalize already.

Reply:
Except it does not... Otherwise I would not have created this PR. There doesn't seem to be any relevant code that finalizes the objects in `generic_factory::finalize`, either:

https://github.com/CleverRaven/Cataclysm-DDA/blob/2c5d4f789754f2059a5e029e7603617173b5b75f/src/generic_factory.h#L284-L287

I also debugged the game in gdb with a breakpoint set to `field_type::finalize`, and without  this change the breakpoint did not fire.

Reply:
Sorry, you are correct - I've got confused.